src/binary-reader-linker.cc

```diff
@@ -163,6 +163,11 @@ Result BinaryReaderLinker::OnImportGlobal(uint32_t import_index,
   return Result::Ok;
 }
 
+Result BinaryReaderLinker::OnFunctionCount(uint32_t count) {
+  binary->debug_names.resize(count + binary->function_imports.size());
```

Comment:
What I'm trying to do here is resize to the size of the function index space.

My understanding is that function index space goes from 0 to (local function count + imported function count).   This callback OnFunctionCount, gives us the number of local functions and function_imports records to number of imported functions.